pkg/cmd/reverseproxy/cmd.go

```diff
@@ -0,0 +1,225 @@
+func NewReverseProxyCommand() *cobra.Command {
+	proxyOpts := &reverseProxyOpts{}
+	cmd := &cobra.Command{
+		Use:   "reverse-proxy",
+		Short: "Start Reverse Proxy for Kube Controller Manager and Cluster Policy Controller to be able to serve metrics on the same port\"",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := proxyOpts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := proxyOpts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	proxyOpts.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (r *reverseProxyOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&r.bindAddress, "listen", r.bindAddress, "The ip:port to serve on.")
+	fs.StringVar(&r.certFile, "tls-cert-file", r.certFile, "serving certificate file (defaults to a generated cert)")
+	fs.StringVar(&r.keyFile, "tls-private-key-file", r.keyFile, "serving certificate key file (defaults to a generated cert)")
+
+	fs.StringVar(&r.kcmPort, "kcm-port", r.kcmPort, "Kube Controller Manager port")
+	fs.StringVar(&r.kcmCABundleFile, "kcm-cabundle-file", r.kcmCABundleFile, "Kube Controller Manager CA bundle certificates")
+
+	fs.StringVar(&r.cpcPort, "cpc-port", r.cpcPort, "Cluster Policy Controller port")
+	fs.StringVar(&r.cpcCABundleFile, "cpc-cabundle-file", r.cpcCABundleFile, "Cluster Policy Controller CA bundle certificates")
+	fs.StringVar(&r.cpcPathPrefix, "cpc-url-path-prefix", r.cpcPathPrefix, "URL path prefix for Cluster Policy Controller")
+}
+
+// Validate verifies the inputs.
+func (r *reverseProxyOpts) Validate() error {
+	if len(r.bindAddress) == 0 {
+		return errors.New("missing required flag: --listen")
+	}
+
+	hasCertFile, hasKeyFile := len(r.certFile) == 0, len(r.keyFile) == 0
+	if hasCertFile != hasKeyFile {
+		return errors.New("incorrect flags: eiter both --tls-cert-file and --tls-private-key-file are required for setting serving cert, or none to auto-generate the serving cert")
+	}
+
+	if len(r.kcmPort) == 0 {
+		return errors.New("missing required flag: --kcm-port")
+	}
+	if len(r.kcmCABundleFile) == 0 {
+		return errors.New("missing required flag: --kcm-cabundle-file")
+	}
+
+	if len(r.cpcPort) == 0 {
+		return errors.New("missing required flag: --cpc-port")
+	}
+	if len(r.cpcCABundleFile) == 0 {
+		return errors.New("missing required flag: --cpc-cabundle-file")
+	}
+	if len(r.cpcPathPrefix) == 0 {
+		return errors.New("missing required flag: --cpc-url-path-prefix")
+	}
+	return nil
+}
+
+// Run contains the logic of the proxy command.
+func (r *reverseProxyOpts) Run() error {
+	if len(r.certFile) == 0 || len(r.keyFile) == 0 {
+		err := r.generateSelfSignedCert()
+		if err != nil {
+			return err
+		}
+	}
+
+	kcmURL, err := url.Parse("https://" + net.JoinHostPort("127.0.0.1", r.kcmPort))
+	if err != nil {
+		return err
+	}
+	cpcURL, err := url.Parse("https://" + net.JoinHostPort("127.0.0.1", r.cpcPort))
+	if err != nil {
+		return err
+	}
+
+	kcmProxy, err := newReverseProxy(kcmURL, r.kcmCABundleFile, "")
```

Comment:
I always  have doubts with this, it should be empty or just a single slash "/"?

Reply:
I just started with `""` for simplicity. But I guess `"/"` looks better since we can reuse the same const variable also for registering the handle func. Updated